web/src/components/storage/DASDTable.tsx

```diff
@@ -144,36 +153,40 @@ const Actions = ({ devices, isDisabled }) => {
   );
 };
 
-const filterDevices = (devices, from, to) => {
+const filterDevices = (devices: DASDDevice[], from: string, to: string): DASDDevice[] => {
   const allChannels = devices.map((d) => d.hexId);
   const min = hex(from) || Math.min(...allChannels);
   const max = hex(to) || Math.max(...allChannels);
 
   return devices.filter((d) => d.hexId >= min && d.hexId <= max);
 };
 
-export default function DASDTable({ state, dispatch }) {
+export default function DASDTable() {
+  const devices = useDASDDevices();
+  const { mutate: changeFilter } = useFilterDASDMutation();
+  const { mutate: changeSelected } = useSelectedDASDChange();
+  const { minChannel, maxChannel } = useFilterDASD();
```

Comment:
@teclator 

I know you're gonna taking care of stop using queries for filtering. But just in case it can help, below you have a diff removing them. Not sending a commit to avoid conflicts with changes you're doing.


```diff
diff --git a/web/src/components/storage/DASDTable.tsx b/web/src/components/storage/DASDTable.tsx
index 80beae41..804ac279 100644
--- a/web/src/components/storage/DASDTable.tsx
+++ b/web/src/components/storage/DASDTable.tsx
@@ -19,7 +19,7 @@
  * find current contact information at www.suse.com.
  */
 
-import React, { Dispatch, SetStateAction, useState } from "react";
+import React, { useState } from "react";
 import {
   Button,
   CardBody,
@@ -46,8 +46,6 @@ import { DASDDevice } from "~/types/dasd";
 import { disableDASD, disableDiag, enableDASD, enableDiag, formatDASD } from "~/api/dasd";
 import {
   useDASDDevices,
-  useFilterDASD,
-  useFilterDASDMutation,
   useSelectedDASD,
   useSelectedDASDChange,
 } from "~/queries/dasd";
@@ -161,13 +159,17 @@ const filterDevices = (devices: DASDDevice[], from: string, to: string): DASDDev
   return devices.filter((d) => d.hexId >= min && d.hexId <= max);
 };
 
+type FilterOptions = {
+  minChannel?: string;
+  maxChannel?: string;
+}
+
 export default function DASDTable() {
   const devices = useDASDDevices();
-  const { mutate: changeFilter } = useFilterDASDMutation();
   const { mutate: changeSelected } = useSelectedDASDChange();
-  const { minChannel, maxChannel } = useFilterDASD();
   const selectedDevices = useSelectedDASD();
 
+  const [{ minChannel, maxChannel }, setFilters] = useState<FilterOptions>({ minChannel: "", maxChannel: "" });
   const [sortingColumn, setSortingColumn] = useState(columns[0]);
   const [sortDirection, setSortDirection] = useState<"asc" | "desc">("asc");
 
@@ -201,22 +203,9 @@ export default function DASDTable() {
     };
   };
 
-  // Filtering
-  const onMinChannelFilterChange = (_event, value) => {
-    changeFilter({ minChannel: value });
-  };
-
-  const onMaxChannelFilterChange = (_event, value) => {
-    changeFilter({ maxChannel: value });
-  };
-
-  const removeMinChannelFilter = () => {
-    changeFilter({ minChannel: "" });
-  };
-
-  const removeMaxChannelFilter = () => {
-    changeFilter({ maxChannel: "" });
-  };
+  const updateFilter = (newFilters: FilterOptions) => {
+    setFilters((currentFilters) => ({ ...currentFilters, ...newFilters }));
+  }
 
   const Content = () => {
     return (
@@ -274,14 +263,14 @@ export default function DASDTable() {
                     type="text"
                     aria-label={_("Filter by min channel")}
                     placeholder={_("Filter by min channel")}
-                    onChange={onMinChannelFilterChange}
+                    onChange={(_, minChannel) => updateFilter({ minChannel })}
                   />
                   {minChannel !== "" && (
                     <TextInputGroupUtilities>
                       <Button
                         variant="plain"
                         aria-label={_("Remove min channel filter")}
-                        onClick={removeMinChannelFilter}
+                        onClick={() => updateFilter({ minChannel: "" })}
                       >
                         <Icon name="backspace" size="s" />
                       </Button>
@@ -296,14 +285,14 @@ export default function DASDTable() {
                     type="text"
                     aria-label={_("Filter by max channel")}
                     placeholder={_("Filter by max channel")}
-                    onChange={onMaxChannelFilterChange}
+                    onChange={(_, maxChannel) => updateFilter({ maxChannel })}
                   />
                   {maxChannel !== "" && (
                     <TextInputGroupUtilities>
                       <Button
                         variant="plain"
                         aria-label={_("Remove max channel filter")}
-                        onClick={removeMaxChannelFilter}
+                        onClick={() => updateFilter({ maxChannel: "" })}
                       >
                         <Icon name="backspace" size="s" />
                       </Button>
diff --git a/web/src/queries/dasd.ts b/web/src/queries/dasd.ts
index f629fdc2..bd07a995 100644
--- a/web/src/queries/dasd.ts
+++ b/web/src/queries/dasd.ts
@@ -171,39 +171,6 @@ const useSelectedDASDChange = () => {
   return mutation;
 };
 
-/**
- * Returns DASD filters
- */
-const filterDASDQuery = () => ({
-  queryKey: ["dasd", "filter"],
-  queryFn: () => {
-    return Promise.resolve({ minChannel: "", maxChannel: "" });
-  },
-  staleTime: Infinity,
-});
-
-const useFilterDASD = (): FilterDASD => {
-  const { data } = useQuery(filterDASDQuery());
-
-  return data || { minChannel: "", maxChannel: "" };
-}
-
-const useFilterDASDMutation = () => {
-  const queryClient = useQueryClient();
-
-  const mutation = useMutation({
-    mutationFn: (data: FilterDASD): Promise<FilterDASD> => Promise.resolve(data),
-    onSuccess: (data: FilterDASD) => {
-      queryClient.setQueryData(["dasd", "filter"], (prev: FilterDASD) => ({
-        ...prev,
-        ...data
-      }));
-    },
-  });
-
-  return mutation;
-};
-
 /**
  * Listens for DASD devices changes.
  */
@@ -255,6 +222,6 @@ const useDASDDevicesChanges = () => {
 };
 
 export {
-  useDASDDevices, useDASDDevicesChanges, useFilterDASDMutation, filterDASDQuery, useFilterDASD, useSelectedDASD, useSelectedDASDChange, selectedDASDQuery,
+  useDASDDevices, useDASDDevicesChanges, useSelectedDASD, useSelectedDASDChange, selectedDASDQuery,
   useDASDFormatJobChanges, useDASDRunningFormatJobs, useFormatDASDMutation
 };
```